src/graphql/device/TypeDefs.js

```diff
@@ -0,0 +1,91 @@
+const TypeDefs = [`
+    enum Type{
+        dynamic
+        static
+        actuate
+    }
+    enum ValueType {
+        integer
+        float
+        text
+        number
+        string
+        boolean
+        geo
+        undefined
+    }
+    type Meta{
+        label: String!
+        desc: String
+        valueType: String!
+        value: String!
+    }
+    type Attr{
+        id: Int!
+        label: String!
+        type: Type!
+        value_type: ValueType!
+        value: String
+        meta: [Meta!]
+    }
+    type Device {
+        #Device id
+        id: String!
+        #Device label
+        label: String!
+        #Attributes
+        attrs: [Attr!]
+        #Template
+        templates: [Template!]!
+    }
+    #Paging things
+    type DeviceListPage {
```

Comment:
# A paginated list of Devices.
type DeviceListPage {
   # Identifies the total count of item.
   totalPages: Int!
   # Indicate the current Page.
   currentPage: Int!
   # List of Devices.
   devices: [Device]
}